src/games/strategy/triplea/ai/proAI/data/ProTerritoryManager.java

```diff
@@ -1,31 +1,1144 @@
+import games.strategy.triplea.ai.proAI.util.ProMatches;
+import games.strategy.triplea.ai.proAI.util.ProTransportUtils;
+import games.strategy.triplea.ai.proAI.util.ProUtils;
+import games.strategy.triplea.attatchments.TerritoryAttachment;
+import games.strategy.triplea.attatchments.UnitAttachment;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.MoveValidator;
+import games.strategy.triplea.delegate.TerritoryEffectHelper;
+import games.strategy.triplea.delegate.TransportTracker;
+import games.strategy.util.CompositeMatchAnd;
+import games.strategy.util.CompositeMatchOr;
+import games.strategy.util.Match;
 
 import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
 
+/**
+ * Manages info about territories.
+ */
 public class ProTerritoryManager {
 
-  final Map<Territory, ProTerritory> territoryMap;
+  private final PlayerID player;
 
-  public ProTerritoryManager(final Map<Territory, ProTerritory> territoryMap) {
-    this.territoryMap = territoryMap;
+  private ProMyMoveOptions attackOptions;
+  private ProMyMoveOptions potentialAttackOptions;
+  private ProMyMoveOptions defendOptions;
+  private ProOtherMoveOptions alliedAttackOptions;
+  private ProOtherMoveOptions enemyDefendOptions;
+  private ProOtherMoveOptions enemyAttackOptions;
+
+  public ProTerritoryManager() {
+    player = ProData.getPlayer();
+    attackOptions = new ProMyMoveOptions();
+    potentialAttackOptions = new ProMyMoveOptions();
+    defendOptions = new ProMyMoveOptions();
+    alliedAttackOptions = new ProOtherMoveOptions();
+    enemyDefendOptions = new ProOtherMoveOptions();
+    enemyAttackOptions = new ProOtherMoveOptions();
+  }
+
+  public ProTerritoryManager(final ProTerritoryManager territoryManager) {
+    this();
+    attackOptions = new ProMyMoveOptions(territoryManager.attackOptions);
+    potentialAttackOptions = new ProMyMoveOptions(territoryManager.potentialAttackOptions);
+    defendOptions = new ProMyMoveOptions(territoryManager.defendOptions);
+    alliedAttackOptions = territoryManager.getAlliedAttackOptions();
+    enemyDefendOptions = territoryManager.getEnemyDefendOptions();
+    enemyAttackOptions = territoryManager.getEnemyAttackOptions();
+  }
+
+  public void populateAttackOptions() {
+    findAttackOptions(player, ProData.myUnitTerritories, attackOptions.getTerritoryMap(),
+        attackOptions.getUnitMoveMap(), attackOptions.getTransportMoveMap(), attackOptions.getBombardMap(),
+        attackOptions.getTransportList(), new ArrayList<Territory>(), new ArrayList<Territory>(),
+        new ArrayList<Territory>(), false, false);
+    alliedAttackOptions = findAlliedAttackOptions(player);
+  }
+
+  public void populatePotentialAttackOptions() {
```

Comment:
Yep. Need to refactor to have get* methods call populate* methods internally when needed. One of the next steps so that the phases only care about getting info while TerritoryManager determines when it needs to calculate.